tests/admin_widgets/test_autocomplete_widget.py

```diff
@@ -0,0 +1,105 @@
+from django import forms
+from django.contrib.admin.widgets import AutocompleteSelect
+from django.forms import ModelChoiceField
+from django.test import TestCase, override_settings
+from django.utils import translation
+
+from .models import Album, Band
+
+
+class AlbumForm(forms.ModelForm):
+    class Meta:
+        rel = Album._meta.get_field('band').remote_field
+        model = Album
+        fields = ['band']
+        widgets = {'band': AutocompleteSelect(Album._meta.get_field('band').remote_field, attrs={'class': 'my-class'})}
+
+
+class NotRequiredBandForm(forms.Form):
+    band = ModelChoiceField(
+        queryset=Album.objects.all(),
+        widget=AutocompleteSelect(Album._meta.get_field('band').remote_field),
+        required=False,
+    )
+
+
+class RequiredBandForm(forms.Form):
+    band = ModelChoiceField(
+        queryset=Album.objects.all(),
+        widget=AutocompleteSelect(Album._meta.get_field('band').remote_field),
+        required=True,
+    )
+
+
+@override_settings(ROOT_URLCONF='admin_widgets.urls')
+class AutocompleteMixinTests(TestCase):
+    def test_build_attrs(self):
+        form = AlbumForm()
+        attrs = form['band'].field.widget.get_context(name='my_field', value=None, attrs={})['widget']['attrs']
+        self.assertIn('class', attrs)
```

Comment:
Is there a reason not to use `assertEqual` with the entire expected dictionary rather than checking bits and pieces?